pkg/deploy/migration/on-reconcile-one-time-migration.go

```diff
@@ -0,0 +1,286 @@
+			return err
+		}
+	}
+
+	// Proxy credentials
+	if ctx.CheCluster.Spec.Server.ProxySecret != "" {
+		if err := addPartOfCheLabelToSecret(ctx, ctx.CheCluster.Spec.Server.ProxySecret); err != nil {
+			return err
+		}
+	}
+
+	// Database credentials
+	if ctx.CheCluster.Spec.Database.ChePostgresSecret != "" {
+		if err := addPartOfCheLabelToSecret(ctx, ctx.CheCluster.Spec.Database.ChePostgresSecret); err != nil {
+			return err
+		}
+	}
+
+	// Keycloak related secrets
+	if ctx.CheCluster.Spec.Auth.IdentityProviderSecret != "" {
+		if err := addPartOfCheLabelToSecret(ctx, ctx.CheCluster.Spec.Auth.IdentityProviderSecret); err != nil {
+			return err
+		}
+	}
+
+	if ctx.CheCluster.Spec.Auth.IdentityProviderPostgresSecret != "" {
+		if err := addPartOfCheLabelToSecret(ctx, ctx.CheCluster.Spec.Auth.IdentityProviderPostgresSecret); err != nil {
+			return err
+		}
+	}
+
+	// Legacy config map with additional CA certificates
+	if ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName != "" {
+		if err := addPartOfCheLabelToConfigMap(ctx, ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName); err != nil {
+			return err
+		}
+	}
+
+	// Config map with CA certificates for git
+	if ctx.CheCluster.Spec.Server.GitSelfSignedCert {
+		if err := addPartOfCheLabelToConfigMap(ctx, deploy.GitSelfSignedCertsConfigMapName); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func addPartOfCheLabelToSecret(ctx *deploy.DeployContext, secretName string) error {
+	return addPartOfCheLabelToObject(ctx, secretName, &corev1.Secret{})
+}
+
+func addPartOfCheLabelToConfigMap(ctx *deploy.DeployContext, configMapName string) error {
+	return addPartOfCheLabelToObject(ctx, configMapName, &corev1.ConfigMap{})
+}
+
+// addPartOfCheLabelToObject adds 'app.kubernetes.io/part-of=che.eclipse.org' label to the object with given name to be cached by operator's k8s client.
+// As the function doesn't know the kind of the object with given name an empty object should be passed,
+// for example: addPartOfCheLabelToObject(ctx, "my-secret", &corev1.Secret{})
+func addPartOfCheLabelToObject(ctx *deploy.DeployContext, objectName string, obj client.Object) error {
+	// Check if the object is already migrated
+	if exists, _ := deploy.GetNamespacedObject(ctx, objectName, obj); exists {
+		// Default client sees the object in cache, no need in adding anything
+		return nil
+	}
+
+	err := ctx.ClusterAPI.NonCachingClient.Get(context.TODO(), types.NamespacedName{Namespace: ctx.CheCluster.Namespace, Name: objectName}, obj)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// The object doesn't exist in cluster, nothing to do
+			return nil
+		}
+		return err
+	}
+
+	// Add part-of labels to the object
+	labels := obj.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	labels[deploy.KubernetesPartOfLabelKey] = deploy.CheEclipseOrg
+	obj.SetLabels(labels)
+	if err := ctx.ClusterAPI.NonCachingClient.Update(context.TODO(), obj); err != nil {
+		return err
+	}
```

Comment:
Duplicates with code below?
Could we log object name and kind?

Reply:
ok